chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+5. **Leverage Modules**: Use the module system to improve code organization
+6. **Document Changes**: Keep track of migration decisions and patterns
+7. **Performance Profiling**: Monitor performance before and after migration
+
+== Migration Guide: from GLSL to Slang
+
+Migrating from GLSL to Slang involves more significant changes than migrating from HLSL, as the languages have different syntax and programming models. However, Slang's powerful features can make the transition worthwhile for many projects.
+
+[TIP]
+====
+For a comprehensive guide on migrating from GLSL to Slang, see the official link:https://docs.shader-slang.org/en/latest/coming-from-glsl.html[Coming from GLSL] documentation.
+====
+
+=== Key Differences to Consider
+
+* *Syntax*: GLSL uses a more procedural style similar to C, while Slang uses an object-oriented style similar to C++
+* *Entry Points*: GLSL uses `void main()`, Slang uses typed functions with explicit inputs/outputs
+* *Resource Binding*: Different approaches to binding resources to shaders
+* *Matrix Layout*: GLSL uses column-major by default, Slang uses row-major by default
+* *Built-ins vs. Semantics*: GLSL uses built-in variables, Slang uses semantics
+
+=== Migration Strategy
+
+1. **Start with a Small Shader**: Begin with a simple shader to understand the migration process
+2. **Understand Semantic Mapping**: Map GLSL built-ins to Slang semantics
+3. **Restructure Resource Bindings**: Convert GLSL uniform blocks to Slang parameter blocks
+4. **Refactor Entry Points**: Convert GLSL's `main()` function to Slang's typed entry points
+5. **Leverage Slang Features**: Gradually introduce Slang-specific features like interfaces and generics
+
+== Slang Compiler
+
+The link:https://github.com/shader-slang/slang[Slang compiler] is used for compiling Slang shaders. The Slang compiler can also compile standard HLSL code, making it a versatile tool for shader development.
+
+=== Where to get
+
+The Slang compiler can be downloaded from the link:https://github.com/shader-slang/slang/releases[official Slang repository]. Pre-built binaries are available for Windows, Linux, and macOS.
+
+=== Offline compilation using the stand-alone compiler
+
+Compiling a Slang shader offline is straightforward:
+
+[source]
+----
+slangc -profile glsl_spirv -entry main -o output.spv input.slang
+----
+
+For HLSL compatibility mode:
+
+[source]
+----
+slangc -profile glsl_spirv -entry main -o output.spv -language hlsl input.hlsl
+----
+
+For GLSL compatibility mode:
+
+[source]
+----
+slangc -profile glsl_spirv -entry main -o output.spv -allow-glsl input.glsl
+----
+
+Key command-line options include:
+
+* `-profile`: Specifies the target profile (use `glsl_spirv` for Vulkan)
+* `-entry`: Specifies the entry point function
+* `-o`: Specifies the output file
+* `-language`: Specifies the input language (default is Slang)
+* `-allow-glsl`: Enables GLSL as an input language
+* `-target`: Specifies the target platform (e.g., `vulkan`)
```

Comment:
There's some sort of a mix-up with `-target` and `-profile` going on here. Typically, `-target` specifies that we want SPIR-V (`-target spirv`) and `-profile` sets the SPIR-V version, e.g. `-profile spirv_1_4`.